android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java

```diff
@@ -278,13 +281,14 @@ private LinearLayout buildRepeatingGroupLayout(ViewParent parent, WidgetArgs wid
         repeatingGroup.addView(repeatingGroupLabel);
 
         JSONArray repeatingGroupJson = new JSONArray(repeatingGroupLayouts.get(((LinearLayout) parent).getId()));
-        String uniqueId = UUID.randomUUID().toString();
+        String groupUniqueId = UUID.randomUUID().toString().replace("-", "");
         for (int i = 0; i < repeatingGroupJson.length(); i++) {
             JSONObject element = repeatingGroupJson.getJSONObject(i);
             String elementType = element.optString(TYPE, null);
             if (elementType != null) {
-                addUniqueIdentifiers(element, uniqueId);
-                FormWidgetFactory factory =  widgetArgs.getFormFragment().getPresenter().getInteractor().map.get(elementType);
+                this.widgetArgs = widgetArgs;
```

Comment:
Wouldn't it be better to assign this in `getViewsFromJson`?

Reply:
Yeah, let me move it......what about the methods where you pass it as an argument since `widgetArgs` will now be global?

Reply:
yes, all method params can now be removed.